services/speech_to_text/v1.js

```diff
@@ -17,54 +17,195 @@
 module.exports = function (RED) {
   var request = require('request');
   var cfenv = require('cfenv');
-  var fs = require('fs');
+  var temp = require('temp');
+  var url = require('url');
+  var fs = require('fs');  
   var fileType = require('file-type');
   var watson = require('watson-developer-cloud');
-  var temp = require('temp');
 
-  temp.track();
+  var service = cfenv.getAppEnv().getServiceCreds(/speech to text/i)
 
-  var username, password;
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then the username and password will be stored in 
+  // the variables sUsername and sPassword. 
+  //
+  // This separation between sUsername and username is to allow 
+  // the end user to modify the credentials when the service is not bound.
+  // Otherwise, once set credentials are never reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
 
-  var service = cfenv.getAppEnv().getServiceCreds(/speech to text/i)
+  var username, password, sUsername, sPassword;
 
   if (service) {
-    username = service.username;
-    password = service.password;
+    sUsername = service.username;
+    sPassword = service.password;
   }
 
+
+  // temp is being used for file streaming to allow the file to arrive so it can be processed. 
+  temp.track();
+
+
+  // These are APIs that the node has created to allow it to dynamically fetch Bluemix
+  // credentials, and also translation models. This allows the node to keep up to 
+  // date with new tranlations, without the need for a code update of this node. 
+
+  // Node RED Admin - fetch and set vcap services
   RED.httpAdmin.get('/watson-speech-to-text/vcap', function (req, res) {
     res.json(service ? {bound_service: true} : null);
   });
 
+
+   // API used by widget to fetch available models
+  RED.httpAdmin.get('/watson-speech-to-text/models', function (req, res) {
+    stt = watson.speech_to_text({
+            username: username ? username : req.query.un,   
+            password: password ? password : req.query.pwd, 
+            version: 'v1',
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 6 space characters but found 12.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636061996/issues/source?bid=3225115&fileBranchId=3319654#l65)